src/WebAuthn.sol

```diff
@@ -146,19 +154,43 @@ library WebAuthn {
         bytes32 clientDataJSONHash = sha256(bytes(webAuthnAuth.clientDataJSON));
 
         // 20. Using credentialPublicKey, verify that sig is a valid signature over the binary concatenation of authData
-        //     and hash.
+        // and hash.
         bytes32 messageHash = sha256(abi.encodePacked(webAuthnAuth.authenticatorData, clientDataJSONHash));
-        bytes memory args = abi.encode(messageHash, webAuthnAuth.r, webAuthnAuth.s, x, y);
-        // try the RIP-7212 precompile address
-        (bool success, bytes memory ret) = _VERIFIER.staticcall(args);
-        // staticcall will not revert if address has no code
-        // check return length
-        // note that even if precompile exists, ret.length is 0 when verification returns false
-        // so an invalid signature will be checked twice: once by the precompile and once by FCL.
-        // Ideally this signature failure is simulated offchain and no one actually pay this gas.
-        bool valid = ret.length > 0;
-        if (success && valid) return abi.decode(ret, (uint256)) == 1;
+        // Attempt RIP-7212 precompile; if ambiguous/false, probe with known-valid vector to
+        // detect precompile presence and avoid unnecessary software fallback when present.
+        if (_rip7212(messageHash, webAuthnAuth.r, webAuthnAuth.s, x, y)) {
+            return true;
+        }
+
+        if (_rip7212(_PROBE_H, uint256(_PROBE_R), uint256(_PROBE_S), uint256(_PROBE_QX), uint256(_PROBE_QY))) {
+            // Precompile is present; original signature invalid.
+            return false;
+        }
 
+        // Precompile absent; fall back to FreshCryptoLib's software verifier.
         return FCL_ecdsa.ecdsa_verify(messageHash, webAuthnAuth.r, webAuthnAuth.s, x, y);
```

Comment:
continue to use FCL instead of OZ verifier because it has slightly better gas performance and was internally audited so we can be confident in the quality of the audit.